OpcUaClient/OpcUaInterface.hpp

```diff
@@ -218,14 +232,193 @@ class OpcUaWrapper : public OpcUaInterface {
+        		if(UA_Variant_hasScalarType(&eventField, &UA_TYPES[UA_TYPES_NODEID])) {
+					UA_NodeId* nodeId = (UA_NodeId *)eventField.data;
+					if(i == 1) {
+						sourceNodeId = nodeId;
+					}
+				}  else {
+					UA_ExtensionObject* extensionObjects = (UA_ExtensionObject*)eventField.data;
+					UA_ModelChangeStructureDataType* modelChangeStructureDataTypes = (UA_ModelChangeStructureDataType*)UA_Array_new(eventField.arrayLength, &UA_TYPES[UA_TYPES_MODELCHANGESTRUCTUREDATATYPE]);
+					for(int j = 0; j < eventField.arrayLength; j ++) {
+						UA_ExtensionObject extensionObject = extensionObjects[j];
+						UA_ModelChangeStructureDataType* modelChangeStructureDataType =(UA_ModelChangeStructureDataType*)extensionObject.content.decoded.data;
+						modelChangeStructureDataTypes[j] = *modelChangeStructureDataType;
+					}
+					OpcUaWrapper* pWrapper = static_cast<OpcUaWrapper*> (monContext);
+					if(pWrapper != nullptr && pWrapper->eventcallback != nullptr) {
+						for(int i = 0; i < eventField.arrayLength; i++) {
+							UA_ModelChangeStructureDataType modelChangeStructureDataType = modelChangeStructureDataTypes[i];
+							UA_NodeId affectedNode = modelChangeStructureDataType.affected;
+							UA_NodeId affectedType = modelChangeStructureDataType.affectedType;
+							UA_Byte verb = modelChangeStructureDataType.verb;
+							Umati::Dashboard::IDashboardDataClient::StructureChangeEvent stc;
+							ModelOpcUa::NodeId_t nodeId = ModelOpcUa::NodeId_t();
+							nodeId.Uri = pWrapper->namespaceArray.at(affectedNode.namespaceIndex);
+							std::string nodeIdPrefix = "i=";
+							switch(affectedNode.identifierType){
+								case UA_NODEIDTYPE_NUMERIC: nodeIdPrefix = "i="; break;
+								case UA_NODEIDTYPE_STRING: nodeIdPrefix = "s="; break;
+								case UA_NODEIDTYPE_BYTESTRING: nodeIdPrefix = "b="; break;
+								case UA_NODEIDTYPE_GUID: nodeIdPrefix = "g="; break;
+							}
+							nodeId.Id = nodeIdPrefix + std::to_string(affectedNode.identifier.numeric);
+							stc.refreshNode = nodeId;
+							stc.nodeAdded = (verb & UA_MODELCHANGESTRUCTUREVERBMASK_NODEADDED) == UA_MODELCHANGESTRUCTUREVERBMASK_NODEADDED;
+							stc.nodeDeleted = (verb & UA_MODELCHANGESTRUCTUREVERBMASK_NODEDELETED) == UA_MODELCHANGESTRUCTUREVERBMASK_NODEDELETED;
+							stc.referenceAdded = (verb & UA_MODELCHANGESTRUCTUREVERBMASK_REFERENCEADDED) == UA_MODELCHANGESTRUCTUREVERBMASK_REFERENCEADDED;
+							stc.referenceDeleted = (verb & UA_MODELCHANGESTRUCTUREVERBMASK_REFERENCEDELETED) == UA_MODELCHANGESTRUCTUREVERBMASK_REFERENCEDELETED;
+							stc.dataTypeChanged = (verb & UA_MODELCHANGESTRUCTUREVERBMASK_DATATYPECHANGED) == UA_MODELCHANGESTRUCTUREVERBMASK_DATATYPECHANGED;
+							pWrapper->eventcallback(stc);
+						}	
+					} else {
+						LOG(ERROR) << "Unable to propagate Event callback!";
+					}
+					UA_Array_delete(modelChangeStructureDataTypes, eventField.arrayLength, &UA_TYPES[UA_TYPES_MODELCHANGESTRUCTUREDATATYPE]);
+				}
+			}
+		}
```

Comment:
There should be some cndition, which might be inverter + continue for less required indentations?